frontend/lib/CompareEndpoints/CompareEndpointsView.dart

```diff
@@ -1,163 +1,129 @@
+                      return Wrap(
+                        children: _createChips(endpointModel),
+                      );
+                    }),
+                    const SizedBox(height: 10),
+                    Consumer<EndpointModel>(
+                      builder: (context, endpointModel, child) {
+                        chart = _createChart(endpointModel);
+                        return chart;
+                      },
+                    ),
+                  ],
+                ),
+              );
+            }
+          },
+        ),
       ),
     );
   }
 
-  Widget _makeInputChips(Future<List<ChartData>> chartDataFuture) {
-    return FutureBuilder<List<ChartData>>(
-        builder: (context, snapshot) {
-          if (snapshot.connectionState == ConnectionState.none ||
-              snapshot.data == null) {
-            return LoadingInCenter();
-          } else {
-        return Wrap(
-                children: snapshot.data!.map(mapChartDataToInputChip).toList());
-          }
-        },
-        future: chartDataFuture);
+  List<Widget> _createChips(EndpointModel endpointModel) {
+    return endpointModel.commonFields.map((e) {
+      endpointModel.selectedChips.putIfAbsent(e, () => false);
+      return Container(
+        margin: const EdgeInsets.symmetric(horizontal: 8, vertical: 8),
+        child: InputChip(
+          label: Text(e),
+          selected: endpointModel.selectedChips[e]!,
+          onSelected: (bool value) {
+            setState(() {
+              endpointModel.selectChips(e, value);
+            });
+          },
+        ),
+      );
+    }).toList();
   }
 
-  Widget mapChartDataToInputChip(ChartData chartData) {
-    isChipsSelected.putIfAbsent(chartData, () => false);
+  Widget _createChart(EndpointModel endpointModel) {
+    if (endpointModel.selectedEndpoints.isEmpty) {
+      return Column();
+    }
+    List<Endpoint> endpoints = endpointModel.getEndpointsForDrawing();
+    List<String> fields = endpointModel.getFieldsForDrawing();
+    List<Widget> charts = [];
 
-    return Container(
-      margin:const EdgeInsets.symmetric(horizontal: 12),
-      child: InputChip(
-        label: Text(chartData.name),
-        selected: isChipsSelected[chartData]!,
-        onSelected: (bool value) {
-          setState(() {
-            isChipsSelected[chartData] = value;
-            if (value) {
-              chartDataSelected.add(chartData);
-            } else {
-              chartDataSelected.remove(chartData);
-            }
-          });
-        },
-      ),
-    );
-  }
+    for (String field in fields) {
+      charts.add(MultiDataChart(field: field, endpoints: endpoints));
+    }
 
-  Iterable<DropdownMenuItem<String>> buildDropDownMenuItems(
-      AsyncSnapshot<List<String>> snapshot) {
-    return snapshot.data!.map((item) => DropdownMenuItem<String>(
-          child: Text(item),
-          value: item,
-        ));
+    return Column(
+      children: charts,
+    );
```

Comment:
W sumie to mogłeś zrobić tak:
```dart
return Column(
      children: fields.map((field) => MultiDataChart(field: field, endpoints: endpoints)).toList(),
    );
```
Chyba krócej będzie tak zapisać, ale to raczej kwestia indywidualna zapisu

Reply:
szczegół, już poprawiam